go/consensus/tendermint/apps/registry/transactions.go

```diff
@@ -573,16 +650,33 @@ func (app *registryApplication) registerRuntime( // nolint: gocyclo
 		if err != nil {
 			return err
 		}
+		// If the governance model is runtime-defined governance, the owner
+		// cannot perform any updates after the initial registration.
+		// Updates to such runtimes should be done via runtime messages.
+		if existingRt.GovernanceModel == registry.GovernanceRuntime {
```

Comment:
If the runtime update message will also call `registerRuntime` then this check will prevent any updates from happening. In case the governance model is the same as in `rt` then this has already been checked in the switch above (checking against `ctx.CallerAddress()`). So the only remaining issue is if the governance model in `rt` would be different than in `existingRt` -- but that should be checked in `VerifyRuntimeUpdate`?

Reply:
You are right, I've fixed this now.  I've also moved the signer checks below this code, so we can do them based on the existing runtime governance model instead of the new one (otherwise it wouldn't be possible to transition from entity to runtime governance, since the transaction for doing that will still be signed with the entity's signer and not the runtime's) :)

Reply:
Looks good now, thanks!